indexing-service/src/test/java/io/druid/indexing/common/task/AppendTaskTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.indexing.common.task;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.druid.indexing.common.TestUtils;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.CountAggregatorFactory;
+import io.druid.segment.IndexSpec;
+import io.druid.timeline.DataSegment;
+import org.joda.time.Interval;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ */
+public class AppendTaskTest
+{
+  @Test
+  public void testSerde() throws Exception
+  {
+    AppendTask appendTask = new AppendTask(
+        "id",
+        "foo",
+        ImmutableList.<DataSegment>of(
+            DataSegment.builder().dataSource("foo").interval(new Interval("2010-01-01/P1D")).version("1234").build()
+        ),
+        ImmutableList.<AggregatorFactory>of(
+            new CountAggregatorFactory("cnt")
+        ),
+        new IndexSpec(),
+        ImmutableMap.<String, Object>of("k", "v")
+    );
+
+    ObjectMapper jsonMapper = new TestUtils().getTestObjectMapper();
+    AppendTask actual = (AppendTask) jsonMapper.readValue(jsonMapper.writeValueAsString(appendTask), Task.class);
+
+    Assert.assertEquals(appendTask, actual);
```

Comment:
this needs to actually check each field, because `equals` on Tasks only compares the id (task ids are assumed unique).